src/tracks/MidiClip.cpp

```diff
@@ -344,6 +344,47 @@ void MidiClip::splitNotes(const NoteVector& notes, TimePos pos)
 	}
 }
 
+void MidiClip::splitNotesAlongLine(TimePos pos1, int key1, TimePos pos2, int key2, bool deleteShortEnds)
+{
+	if (m_notes.empty()) { return; }
+	// Don't split if the line is horitzontal
+	if (key1 == key2) { return; }
+
+	addJournalCheckPoint();
+
+	NoteVector notesCopy = m_notes;
+
+	float slope = 1.f * (pos2 - pos1) / (key2 - key1);
+
+	for (const auto& note : notesCopy)
+	{
+		if (note->key() < std::min(key1, key2) || note->key() > std::max(key1, key2)) { continue; }
+
+		TimePos keyIntercept = slope * (note->key() - key1) + pos1;
+		if (note->pos() < keyIntercept && note->endPos() > keyIntercept)
+		{
+			Note newNote1 = Note(*note);
+			newNote1.setLength(keyIntercept - note->pos());
+
+			Note newNote2 = Note(*note);
+			newNote2.setPos(keyIntercept);
+			newNote2.setLength(note->endPos() - keyIntercept);
+
+			if (deleteShortEnds)
+			{
+				if (newNote1.length() >= newNote2.length()) { addNote(newNote1, false); }
+				else { addNote(newNote2, false); }
+			}
+			else
+			{
+				addNote(newNote1, false);
+				addNote(newNote2, false);
+			}
+
+			removeNote(note);
+		}
+	}
```

Comment:
I made some changes. In particular, I moved the minmax calulation outside the loop and simplified some branches. Feel free to double check my work and add it in if you'd like.

Reply:
Agh, I hate this vscode extension... sorry